frontend/www/js/omegaup/components/arena/ContestListv2.test.ts

```diff
@@ -235,6 +235,64 @@ describe('ContestListv2.vue', () => {
     expect(pastContestTab.text()).toContain('Past Contest 1');
   });
 
+  it('Should handle filter buttons', async () => {
+    const wrapper = mount(arena_ContestList, {
+      propsData: {
+        contests,
+        tab: ContestTab.Current,
+      },
+    });
+
+    const dropdownFilterBy = wrapper.findComponent({
+      ref: 'dropdownFilterBy',
+    });
+
+    // Current filter  "By All" is turned on by default
+    expect(wrapper.vm.currentFilterByAll).toBe(true);
+
+    // When this filter is unchecked, all filters are turned off
+    await dropdownFilterBy.find('[data-filter-by-all]').trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(false);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(false);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(false);
+
+    // When this filter is checked again, all filters are turned on
+    await dropdownFilterBy.find('[data-filter-by-all]').trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(true);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(true);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(true);
+
+    await dropdownFilterBy.find('[data-filter-by-signed-up]').trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(false);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(false);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(true);
+
+    await dropdownFilterBy
+      .find('[data-filter-by-recommended]')
+      .trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(false);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(false);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(false);
+
+    await dropdownFilterBy.find('[data-filter-by-signed-up]').trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(false);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(true);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(false);
+
+    await dropdownFilterBy
+      .find('[data-filter-by-recommended]')
+      .trigger('click');
+
+    expect(wrapper.vm.currentFilterByAll).toBe(true);
+    expect(wrapper.vm.currentFilterBySignedUp).toBe(true);
+    expect(wrapper.vm.currentFilterByRecommended).toBe(true);
+  });
```

Comment:
The test shoul be replaced with the following:
```suggestion
  it('Should handle filter buttons', async () => {
    const wrapper = mount(arena_ContestList, {
      propsData: {
        contests,
        tab: ContestTab.Current,
      },
    });

    const dropdownFilterBy = wrapper.findComponent({
      ref: 'dropdownFilterBy',
    });

    // Current filter "By All" is turned on by default
    expect(wrapper.vm.currentFilter).toBe(ContestFilter.All);

    await dropdownFilterBy.find('[data-filter-by-signed-up]').trigger('click');

    expect(wrapper.vm.currentFilter).toBe(ContestFilter.SignedUp);

    await dropdownFilterBy
      .find('[data-filter-by-recommended]')
      .trigger('click');

    expect(wrapper.vm.currentFilter).toBe(ContestFilter.OnlyRecommended);
  });
```
Note that you are using the enum `ContestFilter`, so it would be necesary import it in this file.